### PR TITLE
Inline mirror://hackage and update URLs

### PR DESCRIPTION
**Pull request by mgorny**
Since `mirror://hackage` has only one URL, inline it in all ebuilds. While at it, also update to the modern hackage URLs (the old ones have permanent redirect to the new ones).

### CODE REVIEW

**Comment by mgorny** [comment]
In case anybody asked, I've verified the new URIs for all packages. There are 404s on other URIs in those packages but hackage URIs are all fine.

**Comment by mgorny** [comment]
@gentoo/haskell, ping.

**Comment by mgorny** [comment]
(I'll rebase once I get an ack)

File: dev-lang/ghc/ghc-7.10.3.ebuild [hunk 1]
```diff
@@ -18,23 +18,23 @@ inherit autotools bash-completion-r1 eutils flag-o-matic ghc-package
...
-arch_binaries="$arch_binaries sparc? ( https://slyfox.uni.cx/~slyfox/distfiles/ghc-bin-${PV}-sparc.tbz2 )"
-arch_binaries="$arch_binaries x86? ( https://slyfox.uni.cx/~slyfox/distfiles/ghc-bin-${PVR}-x86.tbz2 )"
+arch_binaries="$arch_binaries alpha? ( http://slyfox.uni.cx/~slyfox/distfiles/ghc-bin-${PV}-alpha.tbz2 )"
+#arch_binaries="$arch_binaries arm? ( http://slyfox.uni.cx/~slyfox/distfiles/ghc-bin-${PV}-arm.tbz2 )"
+arch_binaries="$arch_binaries amd64? ( http://slyfox.uni.cx/~slyfox/distfiles/ghc-bin-${PVR}-amd64.tbz2 )"
+arch_binaries="$arch_binaries ia64?  ( http://slyfox.uni.cx/~slyfox/distfiles/ghc-bin-${PV}-ia64.tbz2 )"
+arch_binaries="$arch_binaries ppc? ( http://slyfox.uni.cx/~slyfox/distfiles/ghc-bin-${PV}-ppc.tbz2 )"
+arch_binaries="$arch_binaries ppc64? ( http://slyfox.uni.cx/~slyfox/distfiles/ghc-bin-${PV}-ppc64.tbz2 )"
+arch_binaries="$arch_binaries sparc? ( http://slyfox.uni.cx/~slyfox/distfiles/ghc-bin-${PV}-sparc.tbz2 )"
+arch_binaries="$arch_binaries x86? ( http://slyfox.uni.cx/~slyfox/distfiles/ghc-bin-${PVR}-x86.tbz2 )"
```
Comment (trofi) [inline dev-lang/ghc/ghc-7.10.3.ebuild]: This does the reverse from what is stated in commit message. `https://slyfox.uni.cx` should just work.

File: app-admin/haskell-updater/haskell-updater-1.2.10.ebuild [hunk 2]
```diff
@@ -8,7 +8,7 @@ inherit eutils haskell-cabal
 
 DESCRIPTION="Rebuild Haskell dependencies in Gentoo"
 HOMEPAGE="https://wiki.haskell.org/Gentoo#haskell-updater"
-SRC_URI="mirror://hackage/packages/archive/${PN}/${PV}/${P}.tar.gz"
+SRC_URI="https://hackage.haskell.org/package/${PN}-${PV}/${P}.tar.gz"
```
Comment (trofi) [inline app-admin/haskell-updater/haskell-updater-1.2.10.ebuild]: `${PN}-${PV}/${P}.tar.gz` looks confusing here. I don't see why not `${P}/${P}.tar.gz`

File: dev-lang/ghc/ghc-7.10.3.ebuild [hunk 3]
```diff
@@ -182,7 +182,7 @@ update_SRC_URI() {
 		set -- $p
 		pn=$1 pv=$2
 
-		SRC_URI+=" mirror://hackage/package/${pn}/${pn}-${pv}.tar.gz"
+		SRC_URI+=" https://hackage.haskell.org/package/${pn}/${pn}-${pv}.tar.gz"
```
Comment (trofi) [inline dev-lang/ghc/ghc-7.10.3.ebuild]: This is not consistent to other changes. The rest of URL mangling in pull request does the equivalent of `${pn}-${pv}/${pn}-${pv}.tar.gz` and not `${pn}/${pn}-${pv}.tar.gz`. They should agree.

File: dev-lang/ghc/ghc-7.10.3.ebuild (hunk 1 above)
Comment (mgorny) [reply dev-lang/ghc/ghc-7.10.3.ebuild]: I do wonder why I did that. Maybe it had some temporary SSL problems. I'll revert it.

File: app-admin/haskell-updater/haskell-updater-1.2.10.ebuild (hunk 2 above)
Comment (mgorny) [reply app-admin/haskell-updater/haskell-updater-1.2.10.ebuild]: I went for the minimal change in my sed, didn't think of actually replacing that. But'll do.

File: dev-lang/ghc/ghc-7.10.3.ebuild (hunk 3 above)
Comment (mgorny) [reply dev-lang/ghc/ghc-7.10.3.ebuild]: Yeah, my grep+sed wnet for `packages` but this one had `package` instead. Will improve.

**Comment by mgorny** [comment]
I suppose I can commit just the ebuild updates for now and keep the tpm entry until everything else is ready. The main idea is to avoid regressions.

**Comment by mgorny** [comment]
Ok, rebased and fixed all the issues. I'm going to update the commit message tomorrow — obviously, my .bash_history has all the seds prior to the right one but the one that handed the slash-to-dash transition isn't there. Anyway, I'll probably write something better then to cover PN/PV while at it.

**Comment by mgorny** [comment]
Ping.

**Comment by mgorny** [comment]
Oh, I didn't get to do the commit message thing yet.

**Pull request closed without merging**